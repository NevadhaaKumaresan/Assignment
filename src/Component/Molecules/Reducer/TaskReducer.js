/* eslint-disable no-case-declarations */
export const initialState = {
    taskDatas: [],
    pendingCount:0,
    completedcount:0,
    }

export const taskReducer = function (state = initialState, action) {
    switch (action.type) {
      case "DATA":
        return { 
            ...state, 
            taskDatas: action.data,
            pendingCount: action.data.filter( data => data.completed === false).length,
            completedcount: action.data.filter( data => data.completed === true).length,
         }
      case "MOVETOCOMPLETE" :
        let copyOfOriginalArray=[...action.originalData];
        let objIndex = copyOfOriginalArray.findIndex((obj => obj.id === action.dataComplete.id));
        copyOfOriginalArray[objIndex].completed = true;
        return { 
          ...state, 
          taskDatas: copyOfOriginalArray,
          pendingCount: copyOfOriginalArray.filter( data => data.completed === false).length,
          completedcount: copyOfOriginalArray.filter( data => data.completed === true).length,
       }

      case "DELETETASK":
        let deleteArr=[...action.originalData];
        let deleteIndex = deleteArr.findIndex((obj => obj.id === action.deleteData.id));
        deleteArr.splice(deleteIndex, 1);
        return { 
          ...state, 
          taskDatas: deleteArr,
          pendingCount: deleteArr.filter( data => data.completed === false).length,
            completedcount: deleteArr.filter( data => data.completed === true).length,
       }
      default:
        return state;
    }
  };
  