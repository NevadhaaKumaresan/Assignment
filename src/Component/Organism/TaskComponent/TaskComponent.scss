$myColor: whitesmoke;
$myDisplay: flex;
.taskContainer{
    display: $myDisplay;
  }
  
  .taskBody{
    margin: 3%;
  }
  .taskContainerBorder{
    border: 2px solid $myColor;
    padding: 1%;
    margin-right: 1%;
    border-radius: 3%;
    height: 450px;
    overflow-y: scroll;
    background-color: aliceblue;
  }

  .clickPointer{
     cursor: pointer
  }

  .displayHeader{
    display: $myDisplay
  }
  .displayHeaderSize{
    flex-basis: 50em;
    font-weight: bold;
  }