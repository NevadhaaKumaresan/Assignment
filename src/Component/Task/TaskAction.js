import axios from 'axios';

export function loadData() {
    return function(dispatch) {
      return axios.get("https://jsonplaceholder.typicode.com/todos")
        .then(({ data }) => {
        dispatch(setData(data));
      });
    };
  }

export const setData = (data) => {
  return{
    type:"DATA",
    data
   }
}

export const moveToComplete = (originalData, dataComplete) => {
return{
  type:"MOVETOCOMPLETE",
  originalData,
  dataComplete
 }
}

export const deleteTask = (originalData, deleteData) => {
return{
  type:"DELETETASK",
  originalData,
  deleteData
 }
}
