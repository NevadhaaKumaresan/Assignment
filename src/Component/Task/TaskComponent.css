
.taskContainer{
    display: flex;
    margin-top: 5%;
  }
  
  .headingFont{
    font-size: 20px;
    background: whitesmoke;
  }
  
  .taskBody{
    margin: 30px;
  }
  .taskContainerBorder{
    border: 2px solid whitesmoke;
    padding: 1%;
    margin-right: 1%;
    border-radius: 3%;
    height: 450px;
    overflow-y: scroll;
    background-color: aliceblue;
  }

  .clickPointer{
     cursor: pointer
  }

  .circle{
    border-radius: 50%;
    width: 42px;
    height: 39px;
    padding: 5px;
    background: wheat;
    border: 1px solid #000;
    color: #000;
    text-align: center;
    font-size: 14px;
  }

  .boldHeader{
   font-weight: bold;
  }

  .displayHeader{
    display: flex
  }
  .displayHeaderSize{
    flex-basis: 50em;
  }