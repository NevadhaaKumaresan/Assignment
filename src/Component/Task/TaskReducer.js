const initialState = {
taskDatas: [],
}

export const reducer = function (state = initialState, action) {
    switch (action.type) {
      case "DATA":
        return { 
            ...state, 
            taskDatas: action.data,
         }
      case "MOVETOCOMPLETE" :
        let copyOfOriginalArray=[...action.originalData];
        let objIndex = copyOfOriginalArray.findIndex((obj => obj.id === action.dataComplete.id));
        copyOfOriginalArray[objIndex].completed = true;
        return { 
          ...state, 
          taskDatas: copyOfOriginalArray,
       }

      case "DELETETASK":
        let deleteArr=[...action.originalData];
        let deleteIndex = deleteArr.findIndex((obj => obj.id === action.deleteData.id));
        deleteArr.splice(deleteIndex, 1);
        return { 
          ...state, 
          taskDatas: deleteArr,
       }
      default:
        return state;
    }
  };