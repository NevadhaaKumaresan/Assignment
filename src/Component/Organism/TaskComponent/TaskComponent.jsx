// eslint-disable-next-line
import React, { useEffect } from 'react';
import {useStore} from '../../../store';
import './TaskComponent.scss';
import { Row } from 'antd';
import 'antd/dist/antd.css';
import { TableComponent } from '../../Molecules/Table/Table';
import { Header } from '../../Atoms/Header/Header';
import {loadData, moveToComplete, deleteTask} from '../../Molecules/Action/TaskAction';
import {PendingColumns, CompletedColumns} from '../../Molecules/Columns/Columns';

export const TaskComponent = () => {
      const [state, dispatch] = useStore();
      const CategoryArray = [ 
        {key: "1",heading: "Pending Task", complete: false, count: "pendingCount"},
        {key: "2",heading: "Completed Task", complete: true, count: "completedcount"},
        ];
      const { taskDatas } = state; 

      const getTaskList = async () => {
        await dispatch(loadData(dispatch));
      };

      useEffect(() => {
        getTaskList();
      }, []);

      const handleDelete = (data) => {
        dispatch(deleteTask(taskDatas, data));
     }

     const handleComplete = (data) => {
      dispatch(moveToComplete(taskDatas, data));
     }
    
    return(
       <div className="taskBody" role="main">
        <Header/>
        <div className="taskContainer">
            {CategoryArray.map((data) =>
            <div key={data.key} className="taskContainerBorder" tabIndex="0">
            <div key={data.key} className="displayHeader">
            <p key={data.key} className="displayHeaderSize">{data.heading}</p>
            <p key={data.key}>{state[data.count]}</p>
            </div>
            <Row>
            <TableComponent columns={data.complete ? CompletedColumns(handleDelete) : PendingColumns(handleDelete, handleComplete)} taskDatas={taskDatas} action={data.complete ? "completed" : "pending"}/>
            </Row>
          </div>
              )}
        </div>
       </div>
      )
}
