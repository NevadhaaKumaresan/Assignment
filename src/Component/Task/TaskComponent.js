import React, { useEffect } from 'react';
import './TaskComponent.css';
import { Row, Table } from 'antd';
import 'antd/dist/antd.css';
import { DeleteTwoTone, CheckCircleTwoTone } from '@ant-design/icons';

export const TaskComponent = (props) => {

    useEffect(() => {
        props.taskData();
        // eslint-disable-next-line react-hooks/exhaustive-deps
      },[]);  
      const pendingColumns = [
        {
          title: 'Title',
          dataIndex: 'title',
        },
        {
          title: '',
          dataIndex: 'completed',
          render: (completed, obj) => {
            return(
            <CheckCircleTwoTone twoToneColor="#52c41a" onClick={() => handleComplete(obj)}/>
            )
          }
        },
        {
          title: '',
          dataIndex: 'deleteIcon',
          render: (deleteIcon, obj) => {
            return(
            <DeleteTwoTone twoToneColor="#eb2f96"  onClick={ () => handleDelete(obj)}/>
            )
          }
        }
      ];

      const completedColumns = [
        {
          title: 'Title',
          dataIndex: 'title',
        },
        {
          title: '',
          dataIndex: 'deleteIcon',
          render: (deleteIcon, obj) => {
            return(
            <DeleteTwoTone twoToneColor="#eb2f96" onClick={() => handleDelete(obj)}/>
            )
          }
        }
      ];

      const handleDelete = (data) => {
        props.deleteTask(props.taskDatas, data);
     }

     const handleComplete = (data) => {
      props.moveToComplete(props.taskDatas, data)
     }
    
    return(
    <div className="taskBody">
    <div className="headingFont boldHeader">List Of Tasks</div>
    <div className="taskContainer">
    <div className="taskContainerBorder">
      <div className="displayHeader">
        <div className="displayHeaderSize boldHeader">Pending Tasks</div>
        <div className="circle">{props.taskDatas.filter( data => data.completed === false).length}</div>
      </div>
        <Row>
        <Table
        columns={pendingColumns}
        rowKey={record => record.id}
        dataSource={props.taskDatas.filter( data => data.completed === false)}
        pagination={false}
      />
        </Row>

      </div>
    <div className="taskContainerBorder">
      {/* <Row>
        <Col span={20} className="boldHeader">Completed Tasks</Col>
        <Col span={2} className="circle">
            {props.taskDatas.filter( data => data.completed === true).length}
            </Col>
        </Row> */}
         <div className="displayHeader">
        <div className="displayHeaderSize boldHeader">Completed Tasks</div>
        <div className="circle">{props.taskDatas.filter( data => data.completed === true).length}</div>
      </div>
      <Row>
        <Table
        rowKey={record => record.id}
        columns={completedColumns}
        dataSource={props.taskDatas.filter( data => data.completed === true)}
        pagination={false}
      />
        </Row>
        </div>
    </div>
  </div>
      )
}