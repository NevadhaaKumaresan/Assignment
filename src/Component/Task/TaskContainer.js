import { connect } from 'react-redux';
import { TaskComponent } from './TaskComponent';
import {loadData, moveToComplete, deleteTask} from './TaskAction';

const mapStateToProps = state => {
  return {
    taskDatas: state.taskDatas,
  };
};
const mapDispatchToProps = dispatch => {
  return {
    taskData: () => dispatch(loadData()),
    moveToComplete: (orgArray, data) => dispatch(moveToComplete(orgArray, data)),
    deleteTask: (orgArray, data) => dispatch(deleteTask(orgArray, data)),
  }
};
export const TaskContainer = connect(mapStateToProps, mapDispatchToProps)(TaskComponent);