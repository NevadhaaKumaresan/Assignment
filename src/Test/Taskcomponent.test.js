/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import {TaskContainer} from '../Component/Task/TaskContainer';
import {TaskComponent} from '../Component/Task/TaskComponent';
import {reducer} from '../Component/Task/TaskReducer';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';


window.matchMedia = window.matchMedia || function() {
  return {
      matches: false,
      addListener: function() {},
      removeListener: function() {}
  };
};

test('matches snapshot', () => {
  // const middlewares = [thunk];
  // const initialState = {
  //           taskDatas: []
  //       };
  // const mockStore = configureMockStore(middlewares)(initialState);
  
  const taskDatas = [{
    "userId": 1,
    "id": 1,
    "title": "delectus aut autem",
    "completed": false
  },
  {
    "userId": 1,
    "id": 2,
    "title": "quis ut nam facilis et officia qui",
    "completed": true
  }];
  const taskData = () => {
    return taskDatas;
  }
  const tree = renderer.create(<TaskComponent taskDatas={taskDatas} taskData={taskData}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('renders the Todo component with props', () => {
  const middlewares = [thunk];
  const initialState = {
            taskDatas: []
        };
  const mockStore = configureMockStore(middlewares)(initialState);

  const taskDatas = [{
    "userId": 1,
    "id": 1,
    "title": "delectus aut autem",
    "completed": false
  },
  {
    "userId": 1,
    "id": 2,
    "title": "quis ut nam facilis et officia qui",
    "completed": true
  }]

  const taskData = () => {
    return taskDatas;
  }
  render(<Provider store={mockStore}><TaskContainer taskDatas={taskDatas} taskData={taskData}/></Provider>);
});


// For reducer//

describe('reducer', () => {
  const initialState = {
    taskDatas: [],
  };

  test('returns the initial state when an action type is not passed', () => {
    const value = reducer(undefined, {});
    expect(value).toEqual(initialState);
  });
});

describe('Task Reducer', () => {
  const initialState = {
    taskDatas: [],
  };

  test('returns the initial state when an action type is not passed', () => {
    const value = reducer(undefined, {});
    expect(value).toEqual(initialState);
  });

  test("Load DATA", () => {
    const setData = {
      type: "DATA",
      data: [
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        },
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ]
    };
    const Value = reducer(initialState, setData);
  
    expect(Value).toEqual({
      taskDatas: [
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        },
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ],
    });
  });

  test("Move task to Completed task", () => {
    const setData = {
      type: "MOVETOCOMPLETE",
      originalData:[
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        },
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ],
      dataComplete: 
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        }
      
    };
    const Value = reducer(initialState, setData);
  
    expect(Value).toEqual({
      taskDatas: [
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": true
        },
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ],
    });
  });

  test("Delete the task", () => {
    const setData = {
      type: "DELETETASK",
      originalData:[
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        },
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ],
      deleteData: 
        {
          "userId": 1,
          "id": 1,
          "title": "delectus aut autem",
          "completed": false
        }
      
    };
    const Value = reducer(initialState, setData);
  
    expect(Value).toEqual({
      taskDatas: [
        {
          "userId": 1,
          "id": 2,
          "title": "quis ut nam facilis et officia qui",
          "completed": true
        }
      ],
    });
  });
});






