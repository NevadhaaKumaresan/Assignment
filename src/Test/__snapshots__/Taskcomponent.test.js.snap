// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="taskBody"
>
  <div
    className="headingFont boldHeader"
  >
    List Of Tasks
  </div>
  <div
    className="taskContainer"
  >
    <div
      className="taskContainerBorder"
    >
      <div
        className="displayHeader"
      >
        <div
          className="displayHeaderSize boldHeader"
        >
          Pending Tasks
        </div>
        <div
          className="circle"
        >
          1
        </div>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-table-wrapper"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <div
                className="ant-table"
              >
                <div
                  className="ant-table-container"
                >
                  <div
                    className="ant-table-content"
                    onScroll={[Function]}
                    style={Object {}}
                  >
                    <table
                      style={
                        Object {
                          "tableLayout": "auto",
                        }
                      }
                    >
                      <colgroup />
                      <thead
                        className="ant-table-thead"
                      >
                        <tr>
                          <th
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            Title
                          </th>
                          <th
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            
                          </th>
                          <th
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        className="ant-table-tbody"
                      >
                        <tr
                          className="ant-table-row ant-table-row-level-0"
                          data-row-key={1}
                          onClick={[Function]}
                          style={Object {}}
                        >
                          <td
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            delectus aut autem
                          </td>
                          <td
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            <span
                              aria-label="check-circle"
                              className="anticon anticon-check-circle"
                              onClick={[Function]}
                              role="img"
                              tabIndex={-1}
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="check-circle"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                  fill="#52c41a"
                                />
                                <path
                                  d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                                  fill="#f6ffed"
                                />
                                <path
                                  d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                                  fill="#52c41a"
                                />
                              </svg>
                            </span>
                          </td>
                          <td
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            <span
                              aria-label="delete"
                              className="anticon anticon-delete"
                              onClick={[Function]}
                              role="img"
                              tabIndex={-1}
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M292.7 840h438.6l24.2-512h-487z"
                                  fill="#fff0f6"
                                />
                                <path
                                  d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-504-72h304v72H360v-72zm371.3 656H292.7l-24.2-512h487l-24.2 512z"
                                  fill="#eb2f96"
                                />
                              </svg>
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="taskContainerBorder"
    >
      <div
        className="displayHeader"
      >
        <div
          className="displayHeaderSize boldHeader"
        >
          Completed Tasks
        </div>
        <div
          className="circle"
        >
          1
        </div>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-table-wrapper"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <div
                className="ant-table"
              >
                <div
                  className="ant-table-container"
                >
                  <div
                    className="ant-table-content"
                    onScroll={[Function]}
                    style={Object {}}
                  >
                    <table
                      style={
                        Object {
                          "tableLayout": "auto",
                        }
                      }
                    >
                      <colgroup />
                      <thead
                        className="ant-table-thead"
                      >
                        <tr>
                          <th
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            Title
                          </th>
                          <th
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        className="ant-table-tbody"
                      >
                        <tr
                          className="ant-table-row ant-table-row-level-0"
                          data-row-key={2}
                          onClick={[Function]}
                          style={Object {}}
                        >
                          <td
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            quis ut nam facilis et officia qui
                          </td>
                          <td
                            className="ant-table-cell"
                            colSpan={null}
                            rowSpan={null}
                            style={Object {}}
                          >
                            <span
                              aria-label="delete"
                              className="anticon anticon-delete"
                              onClick={[Function]}
                              role="img"
                              tabIndex={-1}
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M292.7 840h438.6l24.2-512h-487z"
                                  fill="#fff0f6"
                                />
                                <path
                                  d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-504-72h304v72H360v-72zm371.3 656H292.7l-24.2-512h487l-24.2 512z"
                                  fill="#eb2f96"
                                />
                              </svg>
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
